packages/js-dpp/lib/dataContract/stateTransition/DataContractUpdateTransition/validation/basic/validateIndicesAreNotChanged.js

```diff
@@ -0,0 +1,56 @@
+const lodashGet = require('lodash.get');
+const DataContractIndicesChangedError = require('../../../../../errors/consensus/basic/dataContract/DataContractIndicesChangedError');
+
+const serializer = require('../../../../../util/serializer');
+
+const ValidationResult = require('../../../../../validation/ValidationResult');
+
+/**
+ * Validate indices have not been changed
+ *
+ * @param {Object} oldDocuments
+ * @param {Object} newDocuments
+ *
+ * @returns {Promise<ValidationResult>}
+ */
+async function validateIndicesAreNotChanged(oldDocuments, newDocuments) {
+  const result = new ValidationResult();
+
+  // Check that old index dinfitions are intact
+  const hasChangedArray = await Promise.all(
+    Object.entries(oldDocuments)
+      .filter(([, schema]) => schema.indices !== undefined)
+      .map(
+        async ([property, oldSchema]) => {
```

Comment:
Why it's async? I feel like logic in this method is overcomplicated. We can just remove `filter` and go through all document type and make sure that indices value is the same (even `undefined`)